homework/index.js

```diff
@@ -1,47 +1,136 @@
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+function renderError(error) {
+  const h2 = document.createElement('h2');
+  document.querySelector('#root').appendChild(h2);
+  h2.innerHTML = error;
+}
+async function fetchJSON(url) {
+  try {
+    const response = await fetch(url);
+    const json = await response.json();
+    return json;
   }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.innerText = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
+  catch (err) {
+    return renderError(err.name);
   }
+}
+
+function createAndAppend(name, parent, options = {}) {
+  const elem = document.createElement(name);
+  parent.appendChild(elem);
+  Object.keys(options).forEach(key => {
+    const value = options[key];
+    if (key === 'text') {
+      elem.innerText = value;
+    } else {
+      elem.setAttribute(key, value);
+    }
+  });
+  return elem;
+}
+
+function renderTableInfo(parent, repositories) {
+  const table = createAndAppend('table', parent);
+  const tbody = createAndAppend('tbody', table);
+  const tr = createAndAppend('tr', tbody);
+  createAndAppend('td', tr, { class: 'label', text: 'Repository : ' });
+  const td = createAndAppend('td', tr);
+  createAndAppend('a', td, {
+    text: repositories.name,
+    href: repositories.html_url,
+    target: '_blank',
+  });
+  const tr1 = createAndAppend('tr', tbody);
+  createAndAppend('td', tr1, { text: 'Description: ', class: 'label' });
+  createAndAppend('td', tr1, {
+    text: repositories.description,
+  });
+  const tr2 = createAndAppend('tr', tbody);
+  createAndAppend('td', tr2, { text: 'Forks :', class: 'label' });
+  createAndAppend('td', tr2, {
+    text: repositories.forks_count,
+  });
+  const dateString = repositories.updated_at;
+  const dateFormat = new Date(dateString);
+  const tr3 = createAndAppend('tr', tbody);
+  createAndAppend('td', tr3, { text: 'Updated :', class: 'label' });
+  createAndAppend('td', tr3, { text: dateFormat.toUTCString() });
+}
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+function listContributors(contributors, parent) {
+  createAndAppend('h3', parent, { class: ' contributor - header', text: 'contributions' });
```

Comment:
There are extraneous spaces in the `class` property value, which probably causes the class not to work. But you don't seem to use this `contributor-header` class in your `style.css` anyway, so best to remove it.